ibis/backends/impala/__init__.py

```diff
@@ -205,7 +204,7 @@ def from_connection(cls, con: hs2.HiveServer2Connection) -> Backend:
     def _post_connect(self) -> None:
         self.options = {}
 
-    @cached_property
+    @property
```

Comment:
for learning purposes, what's the reason of moving away from a cached property. Isn't it `version` something that we would want to avoid executing multiple times?

Reply:
Actually, I changed these based on misunderstanding of `cached_property`. It turns out that it uses regular instance attributes. I thought it was using `lru_cache` and when that's used as a decorator on a method, it leaks memory due to the closed-over cache holding an instance of the class until the interpreter shuts down.

So, the reason is "I didn't understand how `cached_property` worked", but now that I do, I'll revert these.

Reply:
Reverted that change.